ports/psoc6/boards/CY8CPROTO-062-4343W/makefile_mtb.mk

```diff
@@ -84,14 +92,22 @@ mpy_program: $(MPY_MAIN_BUILD_DIR)/firmware.hex
 	$(info Programming done.)
 
 # Use this target to program multiple attached target devices
-mpy_program_multi: 
+mpy_program_multi: attached_devs
 	@:
-	$(eval ATTACHED_TARGET_LIST = $(shell $(MTB_HOME)/tools_3.0/fw-loader/bin/fw-loader --device-list | sed -n -e 's/[0-9]: KitProg3 CMSIS-DAP BULK-//' -e 's/FW Version[^0-9]*\(\([0-9]\.\)\{0,4\}[0-9][^.]\).*//p'| sed -n 's/^[ \t]*//p'))
-	$(eval ATTACHED_TARGETS_NUMBER = $(words $(ATTACHED_TARGET_LIST)))
-	$(info Number of attached targets : $(ATTACHED_TARGETS_NUMBER))
-	$(info List of attached targets : $(ATTACHED_TARGET_LIST))
 	$(foreach ATTACHED_TARGET, $(ATTACHED_TARGET_LIST), $(MAKE) mpy_program SERIAL_ADAPTER_CMD='adapter serial $(ATTACHED_TARGET)';)
 
+mpy_program_ext_hex:
```

Comment:
This is similar to mpy_program right? Where this is useful? I couldn't get it  ..

Reply:
Yes, but while `mpy_program` is depending on the building of the hex file.` mpy_program_ext_hex:` is expected that the .hex is passed as argument, already built.
And the reason to do so is the ci/cd setup, where in one stage we build the binary (on the github ci server), and in the next stage (on the HIL computer) we just flash that .hex. 
This way we don´t have to build in the HIL, and build check can always run in the "cloud" which is faster that our HIL. 

Before we used directly `mpy-psoc6.sh/.cmd` scripts, but this does not support multiple device flashing. And now, for multi tests, we need to flash 2 devices. That requires identifying the kit3prog probes, which needs fw-loader (part of the mtb toolchain). which is used to get the kit3prog serial number (which has to be passed to openocd for selecting the individual devices).
If we add this function to the mpy-psoc6 script we have to also provide that fw-loader tool separately and install it...

So... while now flashing devices simultaneously at user level is no very much needed (and retrieving mtb tools, well you know...) having this option directly in the makefile was the solution I went for. 

Documenting all these makefile targets in a `help` target is in my plans :)

Reply:
Oki....Got it!🎉